rfcs/0007-additional-type-representations.md

```diff
@@ -0,0 +1,48 @@
+# Additional Type Representations
+
+## Purpose
+
+The current list of scalar type representations is inadequate:
+
+- JSON integer and number types are not what we want for most applications, since they are represented by an unusual 53-bit number type. Since scalars can be both inputs and outputs, and are therefore invariant, we cannot safely convert to and from larger and smaller integer and floating point types.
+- We want to be able to generate some standard scalar types in Hasura for types which are used across different data sources, e.g. UUIDs. This needs support in NDC.
```

Comment:
Why does this influence type representations?

As far as I understand the purpose of type representations, they essentially only serve to fill a gap in query/result encoding that is underspecfied without them, i.e. they help answering the question of: "Given some selection field of some scalar type, what kind of literal can I expect in the query result for this field?" and vice versa for input fields. Is this view of their use is too narrow?

If the purpose is to allow e.g. remote relationships between different hosted graphql service over, say, a common `UUID` type, isn't that solved by simply picking the name `UUID` for the scalar type and hoping/asking/ensuring it be represented as a string by the ndc?

Reply:
> Why does this influence type representations?

Good question. Representations were initially intended as a way to communicate hints for code generation, e.g. GraphQL schemas. But now we are planning to generate a fixed set of scalars on the engine side, this is the obvious place to indicate those as well, where it makes sense to do so.

That is arguably violating the "D" in SOLID, since the server spec is now dependent on client details. The alternative is an open world of scalars, in which engine publishes the scalars it supports, and there is some coupling between connectors using those scalars and engine's definition of them. Which arguably violates "I" in SOLID, since servers are forced to implement (parts of) interfaces they may not use.